test/Elastic.Apm.Tests/WildcardMatcherTests.cs

```diff
@@ -5,14 +5,36 @@
 
 using System.Collections.Generic;
 using System.Diagnostics.CodeAnalysis;
+using System.Linq;
 using Elastic.Apm.Helpers;
+using Elastic.Apm.Libraries.Newtonsoft.Json.Linq;
+using Elastic.Apm.Tests.Utilities;
 using FluentAssertions;
 using Xunit;
 
 namespace Elastic.Apm.Tests
 {
 	public class WildcardMatcherTests
 	{
+// Remove warning about unused test parameter "testCaseName"
+#pragma warning disable xUnit1026
+		[Theory]
+		[JsonFileData("./TestResources/json-specs/wildcard_matcher_tests.json")]
+		public void TestCasesFromJsonSpec(string testCaseName, dynamic data)
+		{
+			var testData = (JObject)data;
+			var key = testData.Properties().First();
+			var value = (JObject)key.Last;
+			var matcher = WildcardMatcher.ValueOf(key.Name);
+			foreach (var kvp in value)
```

Comment:
Since Rider shows this on my machine, I'm adding it here:
`value` could technically be `null` - same for `kvp.Value.Value<>` below.

<img width="828" alt="image" src="https://user-images.githubusercontent.com/1091853/194033516-2840adf1-e223-472e-b8ac-219b451c47c9.png">

On the other hand, I don't think we should have totally bulletproof null-checks in test code. I personally think it's much better to have good readability, and if the test code throws nullref exceptions, then we'll look at the test data.

I leave this up to you.

Reply:
If it's fine for you, I'll just leave this as it is.
As you [said in a related PR](https://github.com/elastic/apm-agent-dotnet/pull/1846#discussion_r984407624), the input data is in our control anyway and it would be an easy fix should a future test case yield this problem.